### PR TITLE
#78 Push k8s images to registry instead of .tar procedure

### PR DESCRIPTION
**Pull request by ppxl**

### CODE REVIEW

**Comment by robertauer** [comment]
k8s.mk should also be included in the main Makefile

**Comment by robertauer** [comment]
k8s.mk should also be included in the main Makefile

**Comment by robertauer** [comment]
`make help` raises a lot of errors and warnings

**Comment by robertauer** [comment]
`make help` raises a lot of errors and warnings

File: build/make/k8s.mk [hunk 1]
```diff
@@ -36,6 +40,16 @@ check-k8s-image-env-var:
...
 	@$(call check_defined, ARTIFACT_ID, app/dogu name)
 
+.PHONY: check-etc-hosts
+check-etc-hosts:
+	@grep -E "^.+\s+${K3S_CLUSTER_FQDN}\$$" /etc/hosts > /dev/null || \
+		(echo "Missing /etc/hosts entry for ${K3S_CLUSTER_FQDN}" && exit 1)
+
+.PHONY: check-insecure-cluster-registry
+check-insecure-cluster-registry:
+	@grep "${K3CES_REGISTRY_URL_PREFIX}1" /etc/docker/daemon.json > /dev/null || \
```
Comment (robertauer) [inline build/make/k8s.mk:50]:
```suggestion
	@grep "${K3CES_REGISTRY_URL_PREFIX}" /etc/docker/daemon.json > /dev/null || \
```

**Review by robertauer — CHANGES_REQUESTED** [review]

File: build/make/k8s.mk [hunk 2]
```diff
@@ -51,41 +65,44 @@ k8s-delete: k8s-generate $(K8S_POST_GENERATE_TARGETS) ## Deletes all dogu relate
...
-	@echo "Apply new dogu resources..."
+k8s-apply: k8s-generate $(K8S_POST_GENERATE_TARGETS) ## Applies all generated K8s resources to the current cluster and namespace.
+	@echo "Apply generated K8s resources..."
 	@kubectl apply -f $(K8S_RESOURCE_TEMP_YAML)
 
 ##@ K8s - Docker
 
 .PHONY: docker-build
-docker-build: check-k8s-image-env-var ## Builds the docker image of the k8s app.
+docker-build: check-k8s-image-env-var ## Builds the production docker image of the K8s app.
```
Comment (robertauer) [inline build/make/k8s.mk:84]: To what does `production` refer here? Is there a `development` docker image?

**Review by robertauer — CHANGES_REQUESTED** [review]

File: build/make/k8s.mk [hunk 3]
```diff
@@ -51,41 +65,44 @@ k8s-delete: k8s-generate $(K8S_POST_GENERATE_TARGETS) ## Deletes all dogu relate
...
 	@echo "Building docker image of dogu..."
 	DOCKER_BUILDKIT=1 docker build . -t $(IMAGE)
 
-${K8S_CLUSTER_ROOT}/${ARTIFACT_ID}.tar: check-k8s-artifact-id docker-build	## Saves the current image into a file into the K8s root path to be available on all nodes.
-	@echo "Saving image into k3ces as $(ARTIFACT_ID).tar..."
-	@docker save $(IMAGE) -o $(K8S_CLUSTER_ROOT)/$(ARTIFACT_ID).tar
-	@echo "Done."
+.PHONY: docker-dev-tag
+docker-dev-tag: check-k8s-image-dev-var docker-build ## Tags a Docker image for local K3ces deployment.
+	@echo "Building docker image of dogu..."
```
Comment (robertauer) [inline build/make/k8s.mk:90]: I think this output is not accurate. Seems like it's copied from `docker-build`.

**Review by robertauer — CHANGES_REQUESTED** [review]

File: build/make/k8s.mk (hunk 1 above)
Comment (ppxl) [reply build/make/k8s.mk]: done.

File: build/make/k8s.mk (hunk 2 above)
Comment (ppxl) [reply build/make/k8s.mk]: done.

**Review by robertauer — APPROVED** [review]

**Pull request merged**